src/common/Core/Search/SearchFilter.ts

```diff
@@ -0,0 +1,15 @@
+import { SearchResultItem } from "../SearchResultItem";
+import { fuseJsSearchFilter } from "./fuseJsSearchFilter";
+import { fuzzySortFilter } from "./fuzzySortFilter";
+import { SearchOptions } from "./SearchOptions";
+
+const searchFilters = {
+    "Fuse.js": (options: SearchOptions) => fuseJsSearchFilter(options),
+    "fuzzysort": (options: SearchOptions) => fuzzySortFilter(options),
+};
+
+export type SearchEngineId = keyof typeof searchFilters;
+
+export function searchFilter(options: SearchOptions & { searchEngineId: SearchEngineId }): SearchResultItem[] {
```

Comment:
This is the external facing filter that wraps the underlying ones thats why I marked the other 2 as internal this is what should be used by extensions so the user can choose their prefered filter

Reply:
Ah ok, got it!